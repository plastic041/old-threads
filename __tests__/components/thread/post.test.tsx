import PostContainer from "~/components/thread/post/post";
import { render, screen } from "@testing-library/react";

describe("PostContainer", () => {
  it("renders correctly", () => {
    const post = {
      id: 1,
      thread_id: 1,
      number: 1,
      username: "username",
      body: "body",
      created_at: "2020-01-01T00:00:00.000Z",
    };
    const view = render(<PostContainer post={post} />);
    expect(view).toMatchSnapshot();
  });
});
