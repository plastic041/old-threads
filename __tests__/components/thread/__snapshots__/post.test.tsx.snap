// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContainer renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="flex flex-col rounded bg-white p-4"
      >
        <div
          class="flex flex-row gap-2"
        >
          <a
            class="whitespace-pre text-teal-500 hover:underline"
            href="/#1"
            id="1"
          >
            &gt;&gt;
            1
          </a>
          <span
            class="text-teal-700"
          >
            username
          </span>
          <span
            class="text-teal-700"
          >
            2020-01-01 am 09:00
          </span>
        </div>
        <p
          class="whitespace-pre-wrap text-teal-900"
          style="word-break: keep-all; overflow-wrap: break-word;"
        >
          body
        </p>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="flex flex-col rounded bg-white p-4"
    >
      <div
        class="flex flex-row gap-2"
      >
        <a
          class="whitespace-pre text-teal-500 hover:underline"
          href="/#1"
          id="1"
        >
          &gt;&gt;
          1
        </a>
        <span
          class="text-teal-700"
        >
          username
        </span>
        <span
          class="text-teal-700"
        >
          2020-01-01 am 09:00
        </span>
      </div>
      <p
        class="whitespace-pre-wrap text-teal-900"
        style="word-break: keep-all; overflow-wrap: break-word;"
      >
        body
      </p>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
